client/src/pages/Login.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useEffect, useState } from "react";
+import { fetchJson, fetchText } from "./../utils";
+import { TextInputField } from "./../components/Components";
+import { Container, Grid, Typography, Button } from "@material-ui/core";
+import { IdentityToken, emptyIdentityToken } from "./../interfaces";
+import Logo from "./../img/Logo.png";
+import "./../css/Login.css";
+
+// Applease typescript
+declare const window: any;
+// Grab handle to our microapps js library
+const microapps = window.microapps;
+
+function Login() {
+  const is_web = window.location == window.parent.location;
+  const is_debug = true;
+  const speed = 3;
```

Comment:
speed might not be a good name here as reviewer would have no ideas what speed it is

Reply:
variable name changed for clarity